drivers/CoreLED/tests/CoreLED_test.cpp

```diff
@@ -0,0 +1,61 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "CoreLED.h"
+
+#include "gtest/gtest.h"
+#include "mocks/leka/SPI.h"
+
+using namespace leka;
+
+class CoreLEDTests : public ::testing::Test
+{
+  protected:
+	CoreLEDTests() : coreled(spimock) {}
+
+	// void SetUp() override {}
+	// void TearDown() override {}
+
+	CoreLED coreled;
+	SPIMock spimock;
+};
+
+TEST_F(CoreLEDTests, initialisation)
+{
+	ASSERT_NE(&coreled, nullptr);
+}
+
+TEST_F(CoreLEDTests, turnOff)
+{
+	coreled.turnOff();
+
+	ASSERT_EQ(false, coreled.isOn());
+}
+
+TEST_F(CoreLEDTests, turnOn)
+{
+	coreled.turnOn();
+
+	ASSERT_EQ(true, coreled.isOn());
+}
+
+TEST_F(CoreLEDTests, setImplementedColor)
+{
+	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00, 0x00, 0x00))).Times(2);
+	EXPECT_CALL(spimock, write(testing::ElementsAre(50, 0xFF, 0x00, 0x00))).Times(N_LEDS);
+	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00))).Times(1);
+	auto expected_belt_color = RGB::pure_red;
+	coreled.setColor(expected_belt_color);
+	coreled.showColor();
+}
+
+TEST_F(CoreLEDTests, setUnimplementedBeltColor)
+{
+	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00, 0x00, 0x00))).Times(2);
+	EXPECT_CALL(spimock, write(testing::ElementsAre(50, 0xFF, 0xFF, 0x00))).Times(N_LEDS);
+	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00))).Times(1);
+	auto expected_belt_color = RGB {255, 255, 0};
+	coreled.setColor(expected_belt_color);
+	coreled.showColor();
+}
```

Comment:
Une autre suggestion pour les tests, utilise ce que tu attends pour les EXPECT_CALL, ça sera plus facile pour changer à un seul endroit ce que tu attends et à comprendre qui est qui

```suggestion
TEST_F(CoreLEDTests, setUnimplementedBeltColor)
{
	auto expected_belt_color = RGB {255, 255, 0};

	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00, 0x00, 0x00))).Times(2);
	EXPECT_CALL(spimock, write(testing::ElementsAre(50, expected_belt_color.red, expected_belt_color.green, expected_belt_color.blue))).Times(N_LEDS);
	EXPECT_CALL(spimock, write(testing::ElementsAre(0x00, 0x00))).Times(1);

	coreled.setColor(expected_belt_color);
	coreled.showColor();
}

```